packages/x-components/src/x-modules/tagging/components/__tests__/tagging.spec.ts

```diff
@@ -0,0 +1,144 @@
+
+function renderTagging({
+  template = `<Tagging
+                :consent="consent"
+                :sessionTTLMs="sessionTTLMs"
+                :queryTaggingDebounceMs="queryTaggingDebounceMs"/>`,
+  consent,
+  sessionTTLMs,
+  queryTaggingDebounceMs,
+  snippetConfig = { ...baseSnippetConfig, consent }
+}: RenderTaggingOptions = {}): RenderTaggingAPI {
+  const [, localVue] = installNewXPlugin();
+  snippetConfig = localVue.observable(snippetConfig);
+
+  const wrapper = mount(
+    {
+      components: {
+        Tagging
+      },
+      props: ['consent', 'snippetConfig', 'sessionTTLMs', 'queryTaggingDebounceMs'],
+      template
+    },
+    {
+      provide: {
+        snippetConfig
+      },
+      propsData: {
+        consent,
+        queryTaggingDebounceMs,
+        sessionTTLMs,
+        snippetConfig
+      },
+      localVue
+    }
+  );
+
+  return {
+    wrapper: wrapper.findComponent(Tagging),
+    on: wrapper.vm.$x.on.bind(wrapper.vm.$x)
+  };
+}
+
+describe('testing Tagging component', () => {
+  it('is an XComponent which has an XModule', () => {
+    const { wrapper } = renderTagging();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('tagging');
+  });
+
+  it('emits ConsentProvided when the consent is set using the prop', () => {
+    const { on } = renderTagging({ consent: true });
+    const eventSpy = jest.fn();
+
+    on('ConsentProvided').subscribe(eventSpy);
+
+    expect(eventSpy).toHaveBeenCalledTimes(1);
+    expect(eventSpy).toHaveBeenNthCalledWith(1, true);
+  });
+
+  it('emits SessionDurationProvided when the session duration is set using the prop', () => {
+    const { on } = renderTagging({ sessionTTLMs: 100 });
+    const eventSpy = jest.fn();
+
+    on('SessionDurationProvided').subscribe(eventSpy);
+
+    expect(eventSpy).toHaveBeenCalledTimes(1);
+    expect(eventSpy).toHaveBeenNthCalledWith(1, 100);
+  });
+
+  it('emits QueryTaggingDebounceProvided when the query debounce is set using the prop', () => {
+    const { on } = renderTagging({ queryTaggingDebounceMs: 150 });
+    const eventSpy = jest.fn();
+
+    on('QueryTaggingDebounceProvided').subscribe(eventSpy);
+
+    expect(eventSpy).toHaveBeenCalledTimes(1);
+    expect(eventSpy).toHaveBeenNthCalledWith(1, 150);
+  });
+
+  // eslint-disable-next-line max-len
+  it('emits ConsentProvided with payload false when the consent prop and the snippet config consent are undefined', () => {
+    const { on } = renderTagging({
+      consent: undefined,
+      snippetConfig: {
+        ...baseSnippetConfig,
+        consent: undefined
+      }
+    });
+    const eventSpy = jest.fn();
+
+    on('ConsentProvided').subscribe(eventSpy);
+    expect(eventSpy).toHaveBeenCalledTimes(1);
+    expect(eventSpy).toHaveBeenNthCalledWith(1, false);
+  });
+
+  it('emits ConsentProvided when the consent is set using the snippet config', async () => {
+    const snippet = Vue.observable<SnippetConfig>({
```

Comment:
You can remove this `Vue.observable` as in the render function you'll use the observable again.